articles/ai-services/openai/includes/chatgpt-spring.md

```diff
@@ -0,0 +1,250 @@
+```
+
+The generated files and folders will resemble the following structure:
+
+```
+ai-chat-demo/
+|-- pom.xml
+|-- mvn
+|-- mvn.cmd
+|-- HELP.md
+|-- src/
+    |-- main/
+    |   |-- resources/
+    |   |   |-- application.properties
+    |   |-- java/
+    |       |-- com/
+    |           |-- example/
+    |               |-- aichatdemo/
+    |                   |-- AiChatApplication.java
+    |-- test/
+        |-- java/
+            |-- com/
+                |-- example/
+                    |-- aichatdemo/
+                        |-- AiChatApplicationTests.java
+``` 
+
+## Edit Spring application
+
+
+1. Edit pom.xml file.
+
+    From the root of the project directory, open the *pom.xml* file in your preferred editor or IDE and overwrite the file with following content: 
+
+    ```xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    	<modelVersion>4.0.0</modelVersion>
+    	<parent>
+    		<groupId>org.springframework.boot</groupId>
+    		<artifactId>spring-boot-starter-parent</artifactId>
+    		<version>3.2.0</version>
+    		<relativePath/> <!-- lookup parent from repository -->
+    	</parent>
+    	<groupId>com.example</groupId>
+    	<artifactId>ai-chat-demo</artifactId>
+    	<version>0.0.1-SNAPSHOT</version>
+    	<name>AIChat</name>
+    	<description>Demo project for Spring Boot</description>
+    	<properties>
+    		<java.version>17</java.version>
+    	</properties>
+    	<dependencies>
+    		<dependency>
+    			<groupId>org.springframework.boot</groupId>
+    			<artifactId>spring-boot-starter</artifactId>
+    		</dependency>
+    		<dependency>                 
+    		    <groupId>org.springframework.experimental.ai</groupId>
+    		    <artifactId>spring-ai-azure-openai-spring-boot-starter</artifactId>
+    		    <version>0.7.0-SNAPSHOT</version>
+    		</dependency>
+    		<dependency>
+    			<groupId>org.springframework.boot</groupId>
+    			<artifactId>spring-boot-starter-test</artifactId>
+    			<scope>test</scope>
+    		</dependency>
+    	</dependencies>
+    	<build>
+    		<plugins>
+    			<plugin>
+    				<groupId>org.springframework.boot</groupId>
+    				<artifactId>spring-boot-maven-plugin</artifactId>
+    			</plugin>
+    		</plugins>
+    	</build>
+    	<repositories>
+    		<repository>
+    			<id>spring-snapshots</id>
+    			<name>Spring Snapshots</name>
+    			<url>https://repo.spring.io/snapshot</url>
+    			<releases>
+    				<enabled>false</enabled>
+    			</releases>
+    		</repository>
+    	</repositories>
+    </project>
+    ```
+
+
+1. From the *src/main/java/com/example/aichatdemo* folder, open *AiChatApplication.java* in your preferred editor or IDE and paste in the following code.
+
+    ```java    
+     package com.example.aichatdemo;
```

Comment:
Comment from @mrbullwinkle: 

I am probably just missing something and I am happy to do a 2nd pass where I test things end-to-end, but I am not seeing anywhere in this code/instructions for this doc where the unique Azure OpenAI deployment name for the model is being set/referenced other than mentioning that you need to deploy a model in the pre-reqs.

Without the unique deployment name being referenced I can't see how this code would work. With OpenAI customers only have to specify the model name so you could certainly abstract away setting that value in a simple chat example and just default to having it query gpt-35-turbo, but that isn't possible with Azure OpenAI. By convention we will often name our deployments to match the model name just to make it clear which model you should be using with a given endpoint/API, but we can't assume that a customer has done this and not give them the option to set the deployment name themselves in the code.

I am also not seeing the environment variables for key/endpoint referenced but I realize the AiClient code may be automatically checking for those variables and thus explicitly setting them in this code snippet may not be necessary.

Reply:
@gm2552 Thanks for all your work on this and for the clarifications. For model deployment name we will still need to include some breadcrumbs for the reader to understand how to swap between different model deployments.

At a minimum the code block needs to include a code comment that states this example is hardcoded to a deployment name of "gpt-35-turbo" and have a line commented out that shows how to override the Spring autoconfig and set the model deployment name to something else. 

Some customers don't have the security permissions to deploy models themselves so they are stuck with whatever name their org gave the model when they deployed it.  With the autoconfig set to "gpt-35-turbo" this quickstart will only work if a customer explicitly deployed the model with that name, if they chose any other name this quickstart will fail to run even if they have that underlying model deployed. 

Some customers also have different org wide naming conventions for their model deployments. Once that change is in for the chat and completions quickstarts I think we should be good to go! Thanks again!

Reply:
@mrbullwinkle I believe you can change the model name with an environment variable similar to the endpoint URLs.  I'm testing this out now.  If this works, I will add the appropriate environment variable with a value `gpt-35-turbo` to  the relevant doc section with notes.